src/registers/xcontrol.rs

```diff
@@ -0,0 +1,95 @@
+//! Access to various extended system registers
+use bitflags::bitflags;
+
+/// Extended feature enable mask register
+#[derive(Debug)]
+pub struct XCr0;
+
+bitflags! {
+    /// Configuration flags of the XCr0 register.
+    pub struct XCr0Flags: u64 {
+        /// Enables x87 FPU
+        const X87 = 1;
+        /// Enables 128-bit (legacy) SSE
+        /// Must be set to enable AVX and YMM
+        const SSE = 1<<1;
+        /// Enables 256-bit SSE
+        /// Must be set to enable AVX
+        const YMM = 1<<2;
+        /// When set, PKRU state management is supported by
+        /// ZSAVE/XRSTOR
+        const MPK = 1<<9;
+        /// When set the Lightweight Profiling extensions are enabled
+        const LWP = 1<<62;
+    }
+}
+
+mod x86_64 {
```

Comment:
This needs to be gated on both the `instructions` and `inline_asm` features:

```suggestion
#[cfg(all(feature = "instructions", feature = "inline_asm"))]
mod x86_64 {
```

This is required because the library also needs to be buildable on stable Rust and on non-x86 architectures. Adding this feature gate should fix the CI errors.